### PR TITLE
Feature/HM-41-[Auth] New controller AuthOptions

### PR DESCRIPTION
**Pull request by petarbelchev**
An Auth Options Controller has been created.
The User Roles endpoint has been moved into the Auth Options Controller.

### CODE REVIEW

File: src/Auth/Auth.Api/Auth.Api.csproj [hunk 1]
```diff
@@ -7,15 +7,6 @@
         <RootNamespace>BuildingMarket.$(MSBuildProjectName.Replace(" ", "_"))</RootNamespace>
     </PropertyGroup>
 
-    <ItemGroup>
-        <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="7.0.13" />
```
Comment (rosteslav) [inline src/Auth/Auth.Api/Auth.Api.csproj:11]: ?? You needed that?? didn't you ?? How?

File: src/Auth/Auth.Application/Features/AuthOptions/Queries/GetAllBrokers/GetAllBrokersQueryHandler.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+using MediatR;
+
+namespace BuildingMarket.Auth.Application.Features.AuthOptions.Queries.GetAllBrokers
+{
+    public class GetAllBrokersQueryHandler(IAuthOptionsRepository authOptionsRepository)
+        : IRequestHandler<GetAllBrokersQuery, IEnumerable<BrokerModel>>
+    {
+        private readonly IAuthOptionsRepository _authOptionsRepository = authOptionsRepository;
+
+        public Task<IEnumerable<BrokerModel>> Handle(GetAllBrokersQuery request, CancellationToken cancellationToken)
```
Comment (rosteslav) [inline src/Auth/Auth.Application/Features/AuthOptions/Queries/GetAllBrokers/GetAllBrokersQueryHandler.cs]: Async/await

File: src/Auth/Auth.Infrastucture/Repositories/AuthOptionsRepository.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+        public async Task<IEnumerable<BrokerModel>> GetAllBrokers()
+        {
+            _logger.LogInformation("DB get all brokers");
+
+            try
+            {
+                var query = from user in _context.Users
+                            join userRole in _context.UserRoles on user.Id equals userRole.UserId
+                            join role in _context.Roles on userRole.RoleId equals role.Id
```
Comment (rosteslav) [inline src/Auth/Auth.Infrastucture/Repositories/AuthOptionsRepository.cs]: This is an inner join. Just by joining with Role Broker that will omit and the non-Broker roker. Just join with  _context.Roles.Where(role.Name == UserRoles.Broker).
Test it if it works correctly!

**Review by rosteslav — CHANGES_REQUESTED** [review]

File: src/Auth/Auth.Api/Auth.Api.csproj (hunk 1 above)
Comment (petarbelchev) [reply src/Auth/Auth.Api/Auth.Api.csproj:11]: We inherit it from the Common project.

**Review by rosteslav — APPROVED** [review]

**Pull request merged**